pkg/util/tracingservice/client.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tracingservice
+
+import (
+	"context"
+	"sort"
+
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness/livenesspb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/rpc"
+	"github.com/cockroachdb/cockroach/pkg/rpc/nodedialer"
+	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/quotapool"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing/tracingpb"
+	"github.com/cockroachdb/cockroach/pkg/util/tracingservice/tracingservicepb"
+)
+
+// NodeLiveness is the subset of the interface satisfied by CRDB's node liveness
+// component that the tracing service relies upon.
+type NodeLiveness interface {
+	GetLivenessesFromKV(context.Context) ([]livenesspb.Liveness, error)
+	IsLive(roachpb.NodeID) (bool, error)
+}
+
+// TraceClientDialer can be used to extract recordings from inflight spans for a
+// given traceID, from all nodes of the cluster.
+type TraceClientDialer struct {
+	tracer       *tracing.Tracer
+	dialer       *nodedialer.Dialer
+	nodeliveness NodeLiveness
+}
+
+// NewTraceClientDialer returns a TraceClientDialer.
+func NewTraceClientDialer(
```

Comment:
[nit] tracingservice.NewTraceClientDialer doesn't feel like a good/descriptive name. I'm not sure what's better. Something collector maybe? tracing/service.New() could serve local traces through an RPC, tracing/collector.New() could reach out to all the various tracing/services?

Reply:
Changed the struct to TraceCollector, and the methods to `New()` now that they are in their own packages.